community/server/src/test/java/org/neo4j/server/rest/dbms/UserServiceTest.java

```diff
@@ -200,25 +226,22 @@ public void shouldReturn404WhenChangingPasswordIfDifferentUser() throws Exceptio
     }
 
     @Test
-    public void shouldReturn404WhenChangingPasswordIfUnknownUser() throws Exception
+    public void shouldReturn422WhenChangingPasswordIfUnknownUser() throws Exception
```

Comment:
422 is what we are doing for the other types of illegal requests so you could argue it is less informative (no distinction in error code). We do however still provide exact descriptions in the messages.
I think standardising error message responses across REST and Bolt should be done as a separate PR.